### PR TITLE
docs update

### PR DESCRIPTION
**Pull request by jordens**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,149 @@
...
+Miniconf is designed to be protocol-agnostic. Any means that can receive key-value input from
+some external source can be used to modify values by path.
+
+## Limitations
+Deferred (non-atomic) access to inner elements of some types is not yet supported. This includes:
+* Complex enums (other than [core::option::Option])
+* Tuple structs (other than [Option], [Array])
+
+## Features
+* `mqtt-client` Enabled the MQTT client feature. See the example in [MqttClient].
```
Comment (ryan-summers) [inline README.md:149]: While these links are valid for `rust-doc`, they are not valid markdown. As such, most of the links to rust code in this doc are broken:

image

File: src/iter.rs [hunk 2]
```diff
@@ -16,8 +16,8 @@ pub struct MiniconfIter<M: ?Sized, const L: usize, const TS: usize> {
 
     /// The remaining length of the iterator.
     ///
-    /// It is used to provide `Iterator::size_hint()`
-    /// and may be None to indicate unknown length.
+    /// It is used to provide an exact and trusted `Iterator::size_hint()`.
```
Comment (ryan-summers) [inline src/iter.rs:19]: I'm not sure what "trusted" is supposed to mean. I don't really think it's needed to say that it's correct...

File: README.md (hunk 1 above)
Comment (jordens) [reply README.md:149]: I know. But I think that's a minor issue, less critical than having redundant and divergent docs in README vs the library docstrings. I'm happy to have a slightly ugly README if in turn the content is much better and in sync with the docs.

Comment (ryan-summers) [reply README.md:149]: Fair enough. We can address it later if it becomes annoying

File: src/iter.rs (hunk 2 above)
Comment (jordens) [reply src/iter.rs]: It is important information as this goes beyond the size_hint promises.
It would be `TrustedLen` if that was stable.

**Review by ryan-summers — APPROVED** [review]

**Pull request merged**